ports/raspberrypi/common-hal/analogbufio/BufferedIn.c

```diff
@@ -143,14 +118,36 @@ void common_hal_analogbufio_bufferedin_deinit(analogbufio_bufferedin_obj_t *self
     dma_channel_unclaim(self->dma_chan);
 }
 
-void common_hal_analogbufio_bufferedin_read(analogbufio_bufferedin_obj_t *self) {
+uint32_t common_hal_analogbufio_bufferedin_readinto(analogbufio_bufferedin_obj_t *self, uint8_t *buffer, uint32_t len, uint8_t bytes_per_sample) {
+    // RP2040 Implementation Detail
+    // Fills the supplied buffer with ADC values using DMA transfer.
+    // If the buffer is 8-bit, then values are 8-bit shifted and error bit is off.
+    // If buffer is 16-bit, then values are not shifted and error bit is present.
+    // Number of transfers is always the number of samples which is the array
+    // byte length divided by the bytes_per_sample.
```

Comment:
This comment needs to be revised, I think, because you now normalize to 16 bit, checking the error bit in the process.